app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,206 @@
+    return visibleCheckboxes
+  }
+
+  isCheckboxInView (checkbox) {
+    const initialOptionContainerHeight = this.checkboxesContainer.style.height
+    const optionListOffsetTop = this.checkboxesInnerContainer.offsetTop
+    const distanceFromTopOfContainer = checkbox.offsetTop - optionListOffsetTop
+    return distanceFromTopOfContainer < initialOptionContainerHeight
+  }
+
+  onTextBoxKeyUp (e) {
+    const ENTER_KEY = 13
+    if (e.keyCode === ENTER_KEY) {
+      e.preventDefault()
+    } else {
+      this.filterCheckboxes()
+    }
+  }
+
+  sanitizeText (text) {
+    text = text.replace(/&/g, 'and')
+    text = text.replace(/[’',:–-]/g, '') // remove punctuation characters
+    text = text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // escape special characters
+    return text.trim().replace(/\s\s+/g, ' ').toLowerCase() // replace multiple spaces with one
+  }
+
+  setContainerHeight (height) {
+    this.checkboxesContainer.style.height = height
+  }
+
+  setup () {
+    const MIN_FILTERED_CHECKBOX_LENGTH = 15
+
+    this.setupStatusBox()
+    this.setupHeading()
+    if (this.checkboxes.length >= MIN_FILTERED_CHECKBOX_LENGTH) {
+      this.setupTextBox()
+    }
+    this.setupHeight()
+  }
+
+  setupHeading () {
+    const element = document.createElement('p')
+    element.className = 'app-checkbox-filter__title'
+    element.setAttribute('aria-hidden', 'true')
+
+    this.heading = element
+    this.container.insertBefore(this.heading, this.container.firstChild)
+  }
+
+  setupHeight () {
+    let initialOptionContainerHeight = this.checkboxesContainer.style.height
+    let height = outerHeight(this.checkboxesInnerContainer)
+
+    // check whether this is hidden by progressive disclosure,
+    // because height calculations won't work
+    if (this.checkboxesContainer.offsetParent === null) {
+      initialOptionContainerHeight = 200
+      height = 200
+    }
+
+    // Resize if the list is only slightly bigger than its container
+    if (height < initialOptionContainerHeight + 50) {
+      this.setContainerHeight(height + 1)
+      return
+    }
+
+    // Resize to cut last item cleanly in half
+    const visibleCheckboxes = this.getVisibleCheckboxes()
+    const lastVisibleCheckbox = visibleCheckboxes[visibleCheckboxes.length - 1]
+    const position = lastVisibleCheckbox.parentNode.offsetTop // parent element is relative
+    this.setContainerHeight(position + (lastVisibleCheckbox.height / 1.5))
+  }
+
+  setupStatusBox () {
+    const element = document.createElement('div')
+    element.className = 'govuk-visually-hidden'
+    element.id = this.container.id + '-checkboxes-status'
+    element.setAttribute('role', 'status')
+
+    this.statusBox = element
+    this.updateStatusBox({
+      foundCount: this.getAllVisibleCheckboxes().length,
+      checkedCount: this.getAllVisibleCheckedCheckboxes().length
+    })
+    this.container.insertAdjacentElement('afterend', this.statusBox)
+  }
+
+  setupTextBox () {
```

Comment:
Something in this function doesn't work in IE9, I'll do a bit of investigation

Reply:
I tried using Chrome's IE tab with IE9 standard compatibilty mode and it seems to work. Does the provider have more than 15 subjects related to its courses?